### DISCUSSION TITLE
Get podname's of each stage in workflow

### DISCUSSION THREAD

**Discussion by kollurukailash** [argoproj/argo-workflows#8535] [Q&A] [answered]
A workflow has step1 -> step2 -> step3. When triggered, How to get pod names at each step1, step2, step3.?

**Purpose:**
Currently, all my logs of workflow were saved as artifact in s3. And to get them through rest API
`/artifacts/{namespace}/{name}/{podName}/{artifactName}`.
need the pod name.

@terrytangyuan

**Comment by terrytangyuan** [comment, marked as answer]
See `pod.name`: https://github.com/argoproj/argo-workflows/blob/master/docs/variables.md#containerscript-templates

**Reply by kollurukailash** [reply]
I would like to get the pod name through rest API using the workflow-name.

**Reply by terrytangyuan** [reply]
If you are using v1 pod names, they are the same as the node ID in node statuses. You can get the node statuses from the workflow spec (via GET call).

**Reply by kollurukailash** [reply]
So, we scrape from `/api/v1/workflows/{namespace}/{workflow_name}` right ?

**Reply by terrytangyuan** [reply]
yes

**Reply by wanlxp** [reply]
How should I obtain the pod name when the pod-name version is v2 ?

**Reply by BrunoGrandePhD** [reply]
@wanlxp Given that the `GET` endpoint doesn't return the pod name when using pod names v2, I got around this by creating an output parameter that captures the pod name (see below). This way, the pod name gets included in the `GET` response. 

```
[...]
outputs:
  parameters:
    - name: pod-name
      value: "{{pod.name}}"
```

**Comment by bkanuka** [comment]
Posting here because its related and this came up when I was searching:  The above is correct for getting the _pod_ name, but the _container_ name will be `main` !
